datarepo-clienttests/src/main/java/utils/KubernetesClientUtils.java

```diff
@@ -267,56 +267,54 @@ public static void deleteRandomPod() throws ApiException, IOException {
 
     logger.info("delete random pod: {}", randomPodName);
 
+    deletePod(randomPodName);
+  }
+
+  public static void deletePod(String podNameToDelete) throws ApiException, IOException {
     // known issue with java api "deleteNamespacedPod()" endpoint
     // https://github.com/kubernetes-client/java/issues/252
     // the following few lines were suggested as a workaround
     // https://github.com/kubernetes-client/java/issues/86
     Call call =
         getKubernetesClientCoreObject()
             .deleteNamespacedPodCall(
-                randomPodName, namespace, null, null, null, null, null, null, null);
+                podNameToDelete, namespace, null, null, null, null, null, null, null);
     Response response = call.execute();
     Configuration.getDefaultApiClient()
         .handleResponse(response, (new TypeToken<V1Pod>() {}).getType());
   }
 
   /**
-   * Wait until the size of the replica set matches the specified number of pods. Times out after
-   * {@link KubernetesClientUtils#maximumSecondsToWaitForReplicaSetSizeChange} seconds. Polls in
-   * intervals of {@link KubernetesClientUtils#secondsIntervalToPollReplicaSetSizeChange} seconds.
+   * Wait until the size of the replica set matches the specified number of running pods. Times out
+   * after {@link KubernetesClientUtils#maximumSecondsToWaitForReplicaSetSizeChange} seconds. Polls
+   * in intervals of {@link KubernetesClientUtils#secondsIntervalToPollReplicaSetSizeChange}
+   * seconds.
    *
    * @param deployment the deployment object to poll
    * @param numberOfReplicas the eventual expected size of the replica set
    */
   public static void waitForReplicaSetSizeChange(V1Deployment deployment, int numberOfReplicas)
       throws Exception {
+    logger.debug("waitForReplicaSetSizeChange");
     int pollCtr =
         Math.floorDiv(
             maximumSecondsToWaitForReplicaSetSizeChange, secondsIntervalToPollReplicaSetSizeChange);
 
-    // get the component label from the deployment object
-    // this will be "api" for most cases, since that's what we're interested in scaling.
-    String deploymentComponentLabel = deployment.getMetadata().getLabels().get(componentLabel);
-
-    // loop through the pods in the namespace
-    // find the ones that match the deployment component label (e.g. find all the API pods)
-    long numPods =
-        listPods().stream()
-            .filter(
-                pod ->
-                    deploymentComponentLabel.equals(
-                        pod.getMetadata().getLabels().get(componentLabel)))
-            .count();
-
-    while (numPods != numberOfReplicas && pollCtr >= 0) {
+    // this gives the pod a chance to start changing the size before we start our checks
+    TimeUnit.SECONDS.sleep(5);
```

Comment:
I'd prefer to pull any hard-coded number out into static variables. Maybe you could also get rid of this separate sleep altogether by initializing numPods and numRunningPods either to different numbers (e.g. -1, -2) or to something different than numberOfReplicas (e.g. -1 should do it). Then the while condition would be true and you'd get to the sleep before the first calculations of numPods and numRunningPods.

Reply:
Oh, good idea! Thanks!